### PR TITLE
crypto: add ECDH shared secret support, in memory variant

### PR DESCRIPTION
**Pull request by zelig**
a prerequisite to pss encryption #663 submitted seperately for easier review

### CODE REVIEW

File: pkg/crypto/dh_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	salt := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, salt); err != nil {
+		panic("reading from crypto/rand failed: " + err.Error())
+	}
+	
+	sk0, err := dh0.SharedKey(&key1.PublicKey, salt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sk1, err := dh1.SharedKey(&key0.PublicKey, salt)
```
Comment (janos) [inline pkg/crypto/dh_test.go:41]: Error is not checked.

File: pkg/crypto/dh_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+	key1, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		t.Fatal(err)
+	}
+	dh1 := crypto.NewDH(key1)
+
+	salt := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, salt); err != nil {
+		panic("reading from crypto/rand failed: " + err.Error())
```
Comment (janos) [inline pkg/crypto/dh_test.go:34]: t.Fatal instead of panic.

**Review by janos — COMMENTED** [review]
There are some lint warnings, otherwise looks good.

**Review by janos — APPROVED** [review]
LGTM. 👍

**Review by acud — APPROVED** [review]
lgtm

**Pull request merged**